src/poetry/core/packages/utils/utils.py

```diff
@@ -133,19 +132,19 @@ def is_python_project(path: Path) -> bool:
     return pyproject or setuptools_project
 
 
-def is_archive_file(name: str) -> bool:
+def is_archive_file(name: Path) -> bool:
```

Comment:
```suggestion
def is_archive_file(name: str | Path) -> bool:
```

From an interface's point of view, I think it makes sense to allow plain file names.